src/parser/statement.js

```diff
@@ -303,35 +304,37 @@ pp.parseForStatement = function (node) {
     return this.parseFor(node, init);
   }
 
+  let usedNowKeyword = false;
+  if (this.hasPlugin("lightscript") && (!forAwait)) {
+    // idx, elem, key, or val begins a LS-enhanced loop.
+    if (
+      (this.isContextual("idx") || this.isContextual("elem") || this.isContextual("key") || this.isContextual("val"))
+    ) {
+      return this.parseEnhancedForIn(node);
+    }
+
+    usedNowKeyword = this.match(tt._now);
+  }
+
   const refShorthandDefaultPos = { start: 0 };
   const init = this.parseExpression(true, refShorthandDefaultPos);
   if (this.match(tt._in) || this.isContextual("of")) {
+    if (this.hasPlugin("lightscript") && (!usedNowKeyword)) {
```

Comment:
whoops, I think my comment may have been swallowed... `usedNowKeyword` here should be replaced with `init.isNowAssign`, I think